### PR TITLE
Add helm_chart_branch support to EKS daemon tests

### PR DESCRIPTION
**Pull request by spanaik**
Add helm_chart_branch support to EKS daemon tests

  Pass helm_chart_branch to Terraform modules that declare it (currently
  the OTEL MetricsV2 test modules). Auto-detects the latest release-*
  branch from aws-observability/helm-charts via the GitHub refs API.
  Engineers can override via the helm_chart_branch workflow input to
  test against their own branch.

  Priority: manual input → auto-lookup latest release → fallback to main.
  Existing EKS tests without a helm_chart_branch variable are unaffected.
  
  
  image

### CODE REVIEW

File: .github/workflows/test-artifacts.yml [hunk 1]
```diff
@@ -1198,13 +1208,31 @@ jobs:
             fi
 
             terraform init
+            HELM_BRANCH=""
+            if grep -q 'helm_chart_branch' *.tf 2>/dev/null; then
+              HELM_BRANCH="${HELM_CHART_BRANCH_INPUT}"
+              if [ -z "$HELM_BRANCH" ]; then
+                HELM_BRANCH=$(curl -sL "https://api.github.com/repos/aws-observability/helm-charts/git/matching-refs/heads/release-" \
+                  | jq -r '[.[].ref | ltrimstr("refs/heads/")] | sort | last // empty')
```
Comment (movence) [inline .github/workflows/test-artifacts.yml]: think it's an edge case but `sort | last' would return `4.9` when there is `4.10`

Comment (spanaik) [reply .github/workflows/test-artifacts.yml]: Nice catch, updated.

**Review by sky333999 — APPROVED** [review]

**Review by movence — APPROVED** [review]
